src/core/server/models/story/story.ts

```diff
@@ -911,3 +939,501 @@ export async function markStoryAsUnarchived(
+) {
+  let key = createKey([comment.id, ...comment.ancestorIDs]);
+
+  // key now contains a `.replies` we want to replace with `.status`.
+  key = key.replace(/\.replies$/, ".status");
+
+  const query = { tenantID, id: storyID };
+  const update = {
+    $set: {
+      [key]: comment.status,
+    },
+  };
+  const options = {
+    arrayFilters: createArrayFilters([comment.id, ...comment.ancestorIDs]),
+    returnDocument: "after",
+  };
+
+  const result = await mongo.stories().findOneAndUpdate(query, update, options);
+
+  return result.value;
+}
+
+const VISIBLE_STATUSES = [GQLCOMMENT_STATUS.APPROVED, GQLCOMMENT_STATUS.NONE];
+
+function pruneCommentForVisibleComments(
+  comment: StoryTreeComment,
+  visibleStatuses: GQLCOMMENT_STATUS[],
+  ignoredAuthorIDs: string[]
+) {
+  // If a comment is not visible to the stream, the user cannot
+  // tab/Z_KEY/c-key to it, ignore it and continue on.
+  if (!VISIBLE_STATUSES.includes(comment.status)) {
+    return null;
+  }
+  // Ignore any ignored users, we do not stop to read these comments
+  // as they are not visible to the current user.
+  if (comment.authorID && ignoredAuthorIDs.includes(comment.authorID)) {
+    return null;
+  }
+
+  // Recursively check all our replies the same as we did the root above.
+  // We'll walk down each level and if it's null, we don't include it.
+  const visibleReplies: StoryTreeComment[] = [];
+  for (const reply of comment.replies) {
+    const result = pruneCommentForVisibleComments(
+      reply,
+      visibleStatuses,
+      ignoredAuthorIDs
+    );
+    if (result) {
+      visibleReplies.push(result);
+    }
+  }
+
+  // Return the us + the valid replies under ourself.
+  return {
+    ...comment,
+    replies: visibleReplies,
+  };
+}
+
+// Helper function to iterate over all the linear first array of
+// root comments at the start of the story's comment tree. This is
+// essentially just a filter in a for-loop that starts the recursion
+// down the tree with the above actual recursive op function.
+function pruneTreeForVisibleComments(
+  tree: StoryTreeComment[],
+  visibleStatuses: GQLCOMMENT_STATUS[],
+  ignoredAuthorIDs: string[]
+) {
+  const visibleRootComments: StoryTreeComment[] = [];
+  for (const rootComment of tree) {
+    // Start the recursion
+    const result = pruneCommentForVisibleComments(
+      rootComment,
+      visibleStatuses,
+      ignoredAuthorIDs
+    );
+
+    // If it's null, the comment is ignored or not visible, only
+    // add to visible list if the comment isn't null.
+    if (result) {
+      visibleRootComments.push(result);
+    }
+  }
+
+  return visibleRootComments;
+}
+
+interface FlattenedTreeComment extends StoryTreeComment {
+  rootIndex: number;
+}
+
+function flattenComment(
+  comment: StoryTreeComment,
+  orderBy: GQLCOMMENT_SORT,
+  rootIndex: number,
+  result: FlattenedTreeComment[]
+) {
+  result.push({ ...comment, rootIndex });
+
+  if (orderBy === GQLCOMMENT_SORT.CREATED_AT_ASC) {
```

Comment:
The order checks need to be switched here in `flattenComment` for replies to be included in the correct order.

Reply:
Rejiggered the ordering in the story tree's also fixed up this flattening algo to match what you should see in stream. Should be working now!